### ISSUE TITLE
Repeating button inputs when using multitaps

### ISSUE THREAD

**Issue by vikbez** [FCare/Kronos#367]
There is a problem on multitap games (for example on bomberman), when trying to play a 10-player game, every configured gamepad works but then there is repeat rate on every button of every gamepad rendering the game unable to play. Every input works, but they are like "turbo pads" of the old days.

-- 

Il y a un problème avec les jeux multitaps (par exemple sur Bomberman). Lorsqu'on essaye de jouer à 10 joueurs, chaque manette configurée fonctionne, mais il a alors un taux de répétition sur chaque bouton de chaque manette rendant le jeu inapte à jouer. Chaque bouton fonctionne, mais comme sur les manettes "turbo pads" de l'epoque.

(J'avais ouvert le meme bug sur mametesters il y a ±1 an, et aussi regardé un peu la source pour essayer de fixer mais j'ai pas trouvé)

(J'ouvre le bug ici car vous semblez bien actif sur le projet et j'espere que vous aurez une idée sur comment fixer le soucis !)

**Comment by barbudreadmon** [comment]
Sur quelle version ? Standalone ou libretro ?

**Comment by vikbez** [comment]
Standalone

**Comment by barbudreadmon** [comment]
Pas sûr de comment c'est implémenté sur la version standalone, par contre je sais que çà fonctionne avec la version libretro (testé avec quelques amis).
J'avais fait une ébauche d'implémentation pour la version standalone non-QT dans le PR https://github.com/FCare/Kronos/pull/134 (en me basant sur la version libretro), je pense que le code pourrait être comparé et éventuellement utilisé avec la version QT.

**Comment by vikbez** [comment]
Merci, je vais retester sur la version libretro

**Comment by vikbez** [comment]
Hmmm ca a l'air de faire la meme avec la version libretro (retroarch 1.7.5 + kronos 1.4.5 952f4e8)
Je configure avec player1-2 en multitap+pad et player3-10 en saturn pad. 
Il n'y a rien d'autre a faire ?

**Comment by barbudreadmon** [comment]
> Il n'y a rien d'autre a faire ?

Non, plutôt bizarre ce que tu me dis, je n'ai pas détecté le problème en faisant une partie à 8 il y a quelques mois. Je vais faire quelques tests pour voir si un problème est apparu entre temps.

> kronos 1.4.5 952f4e8

Pourquoi utiliser une version d'il y a plus d'un mois ?

**Comment by vikbez** [comment]
C'est la version qui est installée par retroarch par le online updater.
La j'ai testé sur osx et c'est le dernier bon build disponible.
Je vais re-re-tester dans 1h sur retroarch windows.

**Comment by barbudreadmon** [comment]
> La j'ai testé sur osx et c'est le dernier bon build disponible.

Mes condoléances pour avoir acheté un mac.
Plus sérieusement, vu l'état du support OpenGL sous mac, le port libretro de Kronos (qui dépend beaucoup de l'OpenGL) a pour l'instant été abandonné sur celui-ci. 

En tout cas je confirme ton problème, par contre c'est plutôt bizarre car çà fonctionnait bien il y a quelques mois.

**Comment by barbudreadmon** [comment]
> En tout cas je confirme ton problème, par contre c'est plutôt bizarre car çà fonctionnait bien il y a quelques mois.

Ou alors l'autre théorie, plus plausible après avoir regardé s'il y avait eu des changements là dessus depuis (ce qui n'est pas le cas j'ai l'impression), c'est qu'on ne s'en est même pas rendu compte tellement c'était bien le bordel de toute façon.

**Comment by barbudreadmon** [comment]
Pour future référence, j'ajoute des liens vers l'implémentation multitap de mednafen, qui semble ne pas être affectée par ce problème : 
https://github.com/libretro/beetle-saturn-libretro/blob/master/mednafen/ss/input/multitap.cpp
https://github.com/libretro/beetle-saturn-libretro/blob/master/mednafen/ss/smpc.cpp

**Comment by vikbez** [comment]
> c'etait bien le bordel

De jouer a 8 ? Si oui ca se remarque quand meme les deplacements etant du coup ultra lents.
De plus, quand tu mets un seul multitap ça fonctionne parfaitement bien.

**Comment by barbudreadmon** [comment]
> Si oui ca se remarque quand meme les deplacements etant du coup ultra lents.

Donc il n'y avait pas ce problème, je m'en serai rendu compte.

**Comment by vikbez** [comment]
Sur windows, fail sur le load du bin/cue... Enfin si tu as reussi a reproduire c'est le principal.
